tests/std/tests/P0448R4_spanstream/test.cpp

```diff
@@ -0,0 +1,1027 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <assert.h>
+#include <span>
+#include <spanstream>
+#include <string_view>
+#include <type_traits>
+
+using namespace std;
+
+template <class CharT>
+constexpr auto get_input_array() {
+    if constexpr (is_same_v<CharT, char>) {
+        return "1 2 3 4 5";
+    } else {
+        return L"1 2 3 4 5";
+    }
+}
+
+template <class CharT>
+constexpr auto get_input_view() {
+    if constexpr (is_same_v<CharT, char>) {
+        return "1 2 3 4 5"sv;
+    } else {
+        return L"1 2 3 4 5"sv;
+    }
+}
+
+template <class Spanbuf>
+class test_buf : public Spanbuf {
+public:
+    using Spanbuf::Spanbuf;
+
+    using Spanbuf::eback;
+    using Spanbuf::egptr;
+    using Spanbuf::epptr;
+    using Spanbuf::gptr;
+    using Spanbuf::pbase;
+    using Spanbuf::pptr;
+
+    using Spanbuf::setp;
+
+    using Spanbuf::seekoff;
+    using Spanbuf::seekpos;
+    using Spanbuf::setbuf;
+};
+
+template <class CharT>
+void test_spanbuf() {
+    using test_buf = test_buf<basic_spanbuf<CharT>>;

```

Comment:
I find it confusing to have a typedef that shadows a template. It looks like the least invasive way to avoid this would be to rename the template to `basic_test_buf`.